sound/soc/sof/intel/hda.c

```diff
@@ -1076,6 +1076,13 @@ static irqreturn_t hda_dsp_interrupt_handler(int irq, void *context)
 {
 	struct snd_sof_dev *sdev = context;
 
+	/* Check GIE bit before proceeding */
+	if (!(snd_sof_dsp_read(sdev, HDA_DSP_HDA_BAR, SOF_HDA_INTCTL) &
+	    SOF_HDA_INT_GLOBAL_EN)) {
+		pr_warn_once("%s: spurious interrupt\n", __func__);
+		return IRQ_NONE;
+	}
+
```

Comment:
if you are reading a register while the DSP is off, isn't this going to return 0xFFFFFFFF and that means this test will not help?

Reply:
It will likely 'just' cause page fault at different line.

Reply:
The page fault was not caused by DSP being off as commented here:
https://github.com/thesofproject/linux/pull/4344#discussion_r1190301714

The call trace quoted in this PR as well as the following one pointed to the same cause:
https://github.com/thesofproject/sof/files/12404629/kernel.20230815.205835.87076.0.txt

The patch is trying to avoid the page fault only, it's not a root-cause fix
```
<4>[36962.878771] RAX: 00000000001fffff RBX: ffff9ce686a3c028 RCX: ffffffffc07fdb9
```

The attempt is quite clear in the original version, but updated based reviews.

Reply:
DId some experiments on my board, whether SOF_HDA_INT_GLOBAL_EN is set or not, it does not change the results of snd_sof_dsp_read(sdev, HDA_DSP_BAR, MTL_HFINTIPPTR), it always returns 1800, so using SOF_HDA_INT_GLOBAL_EN wouldn't catch the case.

Reply:
How did you test SOF_HDA_INT_GLOBAL_EN ? SOF_HDA_INT_GLOBAL_EN indicates internal hw is not ready, how do you build this state ?

Reply:
Hi, @RanderWang, I checked MTL_HFINTIPPTR value here:
https://github.com/thesofproject/linux/blob/topic/sof-dev/sound/soc/sof/intel/hda.c#L1216

So my conclusion is that SOF_HDA_INT_GLOBAL_EN is 0 does not necessitate the condition for incorrect MTL_HFINTIPPTR value.

Also since @ujfalusi is working on a solution for this issue, let me close this one.